packages/esm-patient-programs-app/src/programs/programs-form.workspace.tsx

```diff
@@ -74,7 +74,14 @@ const ProgramsForm: React.FC<ProgramsFormProps> = ({
 
   const eligiblePrograms = currentProgram
     ? [currentProgram]
-    : filter(availablePrograms, (program) => !includes(map(enrollments, 'program.uuid'), program.uuid));
+    : filter(
+        availablePrograms,
+        (program) =>
+          !includes(map(enrollments, 'program.uuid'), program.uuid) ||
+          enrollments.some(
+            (enrollment) => enrollment.program.uuid === program.uuid && enrollment.dateCompleted !== null,
+          ),
+      );
```

Comment:
And how about doing this here instead:

```tsx
  const eligiblePrograms = currentProgram
    ? [currentProgram]
    : filter(availablePrograms, (program) => {
        const existingEnrollment = enrollments.find((enrollment) => enrollment.program.uuid === program.uuid);

        return !existingEnrollment || existingEnrollment.dateCompleted !== null;
      });
```

Which cleans up the logic a fair bit and makes it easier to add in custom implementation specific logic in the future for determining eligibility for enrollment.